### ISSUE TITLE
How to login to localhost IMAP account (no domain)

### ISSUE THREAD

**Issue by gitrickm** [RainLoop/rainloop-webmail#1585]
**RainLoop version, browser, OS:**

RainLoop 1.11.3
Debian 9 (stretch)
Apache2, Firefox

**Expected behavior and actual behavior:**

I've been running Linux for 25 years and have never defined a domain name for my desktop. Everything has always worked. I run an IMAP server, but the email is gathered from other servers using fetchmail.

Thunderbird and Squirrelmail can both access the IMAP account with localhost, port 143 and my username and password. There is no mention of a domain name.

**Steps to reproduce the problem:**

I specified "Use short login". I get "Authentication failed" and the log says "The email address "myname" is not complete. I've tried both "localhost" and my IP address 192.168.nnn.nnn.

There is also an error:  InvalidInputArgument[903]

**Logs or screenshots:**

[08:29:09.541][9fa3867d] INFO[DATA]: [DATE:04.12.17][OFFSET:-00][RL:1.11.3][PHP:5.6.30-0+deb8u1][IP:127.0.0.1][PID:2665][Apache/2.4.25 (Debian)][apache2handler]
[08:29:09.541][9fa3867d] INFO[DATA]: [Suhosin:off][APC:off][MB:on][PDO:~][Streams:tcp,udp,unix,udg,ssl,sslv3,tls,tlsv1.0,tlsv1.1,tlsv1.2]
[08:29:09.541][9fa3867d] REQUEST[NOTE]: [POST] http://localhost/rainloop/?/Ajax/&q[]=/0/
[08:29:09.541][9fa3867d] AJAX[NOTE]: Action: DoLogin
[08:29:09.541][9fa3867d] POST[DATA]: {"Email":"myname","Login":"","Password":"*******","Language":"","AdditionalCode":"","AdditionalCodeSignMe":"0","SignMe":"0","Action":"Login","XToken":"34aaf0f2dbd4d56f22a88bbbe3af570a"}
[08:29:09.541][9fa3867d] LOGIN[DATA]: The email address "myname" is not complete
[08:29:10.542][9fa3867d] INFO[NOTICE]: exception 'RainLoop\Exceptions\ClientException' with message 'InvalidInputArgument[903]' in /var/www/html/rainloop/rainloop/v/1.11.3/app/libraries/RainLoop/Actions.php:2228
Stack trace:
#0 /var/www/html/rainloop/rainloop/v/1.11.3/app/libraries/RainLoop/Actions.php(2374): RainLoop\Actions->LoginProcess('myname', '*******', '', '', false)
#1 [internal function]: RainLoop\Actions->DoLogin()
#2 /var/www/html/rainloop/rainloop/v/1.11.3/app/libraries/RainLoop/ServiceActions.php(172): call_user_func(Array)
#3 [internal function]: RainLoop\ServiceActions->ServiceAjax('')
#4 /var/www/html/rainloop/rainloop/v/1.11.3/app/libraries/RainLoop/Service.php(146): call_user_func(Array, '')
#5 /var/www/html/rainloop/rainloop/v/1.11.3/app/libraries/RainLoop/Service.php(56): RainLoop\Service->localHandle()
#6 /var/www/html/rainloop/rainloop/v/1.11.3/app/libraries/RainLoop/Service.php(79): RainLoop\Service->__construct()
#7 /var/www/html/rainloop/rainloop/v/1.11.3/app/handle.php(94): RainLoop\Service::Handle()
#8 /var/www/html/rainloop/rainloop/v/1.11.3/include.php(228): include('/var/www/html/r...')
#9 /var/www/html/rainloop/index.php(13): include('/var/www/html/r...')
#10 {main}
[08:29:10.542][9fa3867d] AJAX[DATA]: {"Action":"Login","Result":false,"ErrorCode":903,"ErrorMessage":"","ErrorMessageAdditional":"","Time":1012}
[08:29:10.544][9fa3867d] INFO[MEMORY]: Memory peak usage: 1024KB
[08:29:10.544][9fa3867d] INFO[TIME]: Time delta: 1.0146820545197

**Comment by gitrickm** [comment]
I upgraded my apache2 instance to use php7.0, but it made no difference to my problem with rainloop. Roundcube also works fine with no domain name, as does Kaiten mail from my Android phone.

I have a gmail account and I was able to login to that, using a full email address. The short login option wouldn't work for me.

I should mention that the "Test" button on the domain page always turns green.

The messages look a bit different now:

[20:41:48.157][3e392763] LOGIN[DATA]: The email address "myname" is not complete
[20:41:48.157][3e392763] LOGIN[DATA]: Determined user domain: localhost
[20:41:48.157][3e392763] LOGIN[DATA]: Check "localhost": NO
[20:41:48.157][3e392763] LOGIN[DATA]: Check "localhost" with wildcard: NO
[20:41:48.157][3e392763] LOGIN[DATA]: Domain was not found!
[20:41:49.158][3e392763] INFO[NOTICE]: RainLoop\Exceptions\ClientException: InvalidInputArgument[903] in /var/www/html/rainloop/rainloop/v/1.11.3/app/libraries/RainLoop/Actions.php:2228
Stack trace:
#0 /var/www/html/rainloop/rainloop/v/1.11.3/app/libraries/RainLoop/Actions.php(2374): RainLoop\Actions->LoginProcess('myname', '*******', '', '', false)
#1 /var/www/html/rainloop/rainloop/v/1.11.3/app/libraries/RainLoop/ServiceActions.php(172): RainLoop\Actions->DoLogin()
#2 /var/www/html/rainloop/rainloop/v/1.11.3/app/libraries/RainLoop/Service.php(146): RainLoop\ServiceActions->ServiceAjax('')
#3 /var/www/html/rainloop/rainloop/v/1.11.3/app/libraries/RainLoop/Service.php(56): RainLoop\Service->localHandle()
#4 /var/www/html/rainloop/rainloop/v/1.11.3/app/libraries/RainLoop/Service.php(79): RainLoop\Service->__construct()
#5 /var/www/html/rainloop/rainloop/v/1.11.3/app/handle.php(94): RainLoop\Service::Handle()
#6 /var/www/html/rainloop/rainloop/v/1.11.3/include.php(228): include('/var/www/html/r...')
#7 /var/www/html/rainloop/index.php(13): include('/var/www/html/r...')
#8 {main}
[20:41:49.158][3e392763] AJAX[DATA]: {"Action":"Login","Result":false,"ErrorCode":903,"ErrorMessage":"","ErrorMessageAdditional":"","Time":1011}

**Comment by ervee** [comment]
Perhaps disable domain autodetection. Create a domain configuration (use your @yourdomain.tld) with short login option. Then login with your email address (name@yourdomain.tld). If the name portion of your email address does not match the IMAP username, use the name mapping plugin.

**Comment by gitrickm** [comment]
I've tried all that. No matter what I configure I either get "Domain is not allowed" or "Authentication failed".

I even added a (fake) domain name to my machine, which I've never had to do for 25 years.

**Comment by ervee** [comment]
Can you post the logs (RainLoop, Apache, Dovecot and perhaps /var/log/auth.log) for when you configure it as in my previous comment?

**Comment by gitrickm** [comment]
OK, I finally got it to work.

No other mail client I've used ever asked for a "domain name", so I thought it was only asking for an "Account Name" for the server (like Thunderbird has, for example). I had "My IMAP". I entered my new fake domain name (myhost.home). The server is simply "localhost" as I had. I had to enable "Use short login" and disable custom-login-mapping. Now it works. In fact, I was able to remove the domain that I defined on my server.

To send mail, I had to use the "php mail() function (beta)". While logged into my account I had to set up a default Identity for my real email address to be used in the From field. There might be a way to configure the SMTP to work but this seems good enough.

So, "my bad", as they say. Thank-you for your interest and help.

I'm glad I have it working now. In some ways it renders better on my phone and tablet than Roundcube does.
